File: sort/Leetcode/javascript/web/js/plusOne.js
const digits = [9,9,9]
//const digits = [1,2,3]
//const digits = [6,1,4,5,3,9,0,1,9,5,1,8,6,7,0,5,5,4,3]

let answer = plusOne(digits);

function plusOne(digits) {
    var stack = [];
    let carry = 1;

    for (let i = digits.length-1; i >= 0; i--) {
        let newNumber = digits[i] + carry;
  
        if(newNumber > 9) { 
            stack.unshift(0);
            carry = 1;
        } else {
            stack.unshift(newNumber);
            carry = 0;
        }
        
        if (i == 0 && carry == 1) { 
            stack.unshift(1);
        }
    }

    return stack;

}


for (let i = 0; i < answer.length; i++) {
    console.log(i + " is " + answer[i]);
}

/*

function plusOne(digits) {
    var answer = []
    for (let i = 0; i < digits.length; i++) {
        //console.log(digits[i]);     
        answer.push(digits[i]);
        
        if(i == digits.length - 1) {
            console.log("this is the last loop " + digits[i])
            if(digits[i] === 9) {
                answer.push(1);
            }
        }
        

   }

    for (let i = digits.length-1; i >= 0; i--) {
         console.log(digits[i]);
    }
    

    return answer;
}


*/



function plusOneNumToBig(digits) {
    var numString = "";
    var num;
    var answer = [];

    for (let i = 0; i < digits.length; i++) {
        numString = numString + digits[i];
        //console.log(digits[i]);
    }
    //console.log(numString)
    
    //6145390195186705543
    //6145390195186705000

    num = parseInt("6145390195186705543");
    console.log(num)
    num = num + 1;
    numString = num.toString();

    for (let i = 0; i < numString.length; i++) {
        answer[i] = numString.charAt(i);
    }

    return answer;
}

File: java/src/main/java/leetcode/PlusOne.java
package leetcode;

import java.util.ArrayList;

public class PlusOne {
    public static void main(String[] args) {
        ArrayList<Integer> digits = new ArrayList<>();
        digits.add(1);
        digits.add(2);
        digits.add(3);
        //Input: digits = [1,2,3]
        //Output: [1,2,4]
        ArrayList<Integer> answer = plusOne(digits);
        System.out.println(answer);
    }

    public static ArrayList<Integer> plusOne(ArrayList<Integer> digits) {
        Integer addAmount = 1;
        Integer carry = 0;

        for (int i = digits.size() - 1; i >= 0; i--) {
            if(i < 10) {
                digits.set(i, digits.get(i) + addAmount + carry);
                carry = 0;
                addAmount = 0;
            } else {
                carry = 1;
                digits.set(i, 0);
                addAmount = 0;
            }

            System.out.println(digits.get(i));
        }


        return digits;
    }
}

/*
        System.out.println(digits);
                Integer carry = 0;

                for (int i = digits.size() - 1; i >= 0; i--) {
                Integer currentNumber = digits.get(i);
                Integer newNumber;

                if(currentNumber + carry < 9) {
        newNumber = currentNumber + 1;
        carry = 0;
        } else {
        newNumber = 0;
        carry = 1;

        }

        digits.set(i, newNumber);

        if(i == 0 && carry == 1) {
        //digits.set(i + 1, carry);
        digits.add(0, carry);
        }
        //Case: Middle or Last Number
        //Case: Less or more then 9
        //Case: Carry or no carry

        //Case 1: Middle Number
        if(i > 0) {
        //System.out.println("middle " + digits.get(i));

        //Case 2: Last Number
        } else {
        //System.out.println("last " + digits.get(i));
        }
        }




        System.out.println(digits);



        for (int i = digits.size() - 1; i >= 0; i--) {
            System.out.println(digits.get(i));

            if(i < 10) {
                digits.set(i, digits.get(i) + 1);
            } else {
                carry = 1;
                digits.set(i, 0);
            }

        }

         */

